pkg/kubelet/rkt/rkt.go

```diff
@@ -1086,12 +1094,43 @@ func (r *Runtime) GetContainerLogs(pod *api.Pod, containerID string, logOptions
 // just GC kubernetes pods.
 func (r *Runtime) GarbageCollect() error {
 	if err := exec.Command("systemctl", "reset-failed").Run(); err != nil {
-		glog.Errorf("rkt: Failed to reset failed systemd services: %v", err)
+		glog.Errorf("rkt: Failed to reset failed systemd services: %v, continue to gc anyway...", err)
 	}
 	if _, err := r.runCommand("gc", "--grace-period="+defaultGracePeriod, "--expire-prepared="+defaultExpirePrepared); err != nil {
-		glog.Errorf("rkt: Failed to gc: %v", err)
+		glog.Errorf("rkt: Failed to gc: %v, continue to gc systemd service files", err)
+	}
+
+	// GC all inactive systemd service files.
+	units, err := r.systemd.ListUnits()
+	if err != nil {
+		glog.Errorf("rkt: Failed to list units: %v", err)
+		return err
+	}
+	runningUnits := sets.NewString()
+	for _, u := range units {
+		if strings.HasPrefix(u.Name, kubernetesUnitPrefix) && u.SubState == "running" {
+			runningUnits.Insert(u.Name)
+		}
+	}
+
+	files, err := ioutil.ReadDir(systemdServiceDir)
+	if err != nil {
+		glog.Errorf("rkt: Failed to read the systemd service directory: %v", err)
 		return err
 	}
+	for _, f := range files {
+		if strings.HasPrefix(f.Name(), kubernetesUnitPrefix) {
```

Comment:
nit: reduce the nested `if` levels by using `continue` here? This is more of a personal preference, so it's up to you.